the-collab-lab/tcl-3-smart-shopping-list
# Issue #1 | Connect app to Firebase

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|  ✓    | :sparkles: New feature     |
|  ✓    | :link: Update dependencies |

## Acceptance Criteria


## Related Issue

Closes #23 

## Description 

This PR adds two new components: one for adding new items to the database and another for displaying the list of items. 

### Display items from database 

Inside that `List` component, we've used `FirestoreCollection` which is a component that allows you to interact with a Firestore collection. Using this component, you can access the collection at a given path and provide sort options, perform queries, and paginate data.

This component will setup a listener and update whenever our collection is updated in Firestore.

More documentation on `FirestoreCollection` [here](https://www.npmjs.com/package/react-firestore#firestorecollection). 

### Write items to the database

When we wrap this `AddItem` component in the higher-order component (HOC) `withFirestore` we're provided some handy dandy methods that make it easy to send documents up to the database.

Check out more documentation and examples of how `withFirestore` is used [here](https://www.npmjs.com/package/react-firestore#withfirestore). 
More documentation on React forms and controlled components [here](https://reactjs.org/docs/forms.html). 


## Testing Steps / QA Criteria

* From your terminal, pull down this branch with `git pull origin stc-connect-firebase` and check that branch out with `git checkout stc-connect-firebase` 
* Then `yarn` to install the newly added dependencies locally and `yarn start` to launch the app.
* Fill out the form with a new item. Make sure it adds it to the list and make sure that item appears in the `items` collection of the database [here](https://console.firebase.google.com/u/0/project/tcl-3-smart-shopping-list/database/firestore/data~2Fitems~2FmXdjt73e7u9HTlvYLl3c).

## TODOs

- [x] Due to the last-minute circumstances that led to the completion of this issue by me (Stacie), tests were not prioritized. If I have time before the tcl-3 demos tomorrow, I will add these. If anyone else sees this and wants to add them: We need to install cypress.io and test the functionality of the form and list.
